src/context/FilterProjectsContext.jsx

```diff
@@ -297,6 +320,30 @@ export const FilterProjectsProvider = ({ children }) => {
             records: filteredRecords,
           }
         })
+        .map((project) => {
+          // Filter by map bounding box
+          if (!enableFollowScreen) {
+            return project
+          }
+
+          return {
+            ...project,
+            records: project.records.filter((record) => {
+              const isWithinLatitude =
+                record.latitude >= mapBbox['_sw'].lat && record.latitude <= mapBbox['_ne'].lat
+
+              const swLng = normalizeLng(mapBbox['_sw'].lng)
+              const neLng = normalizeLng(mapBbox['_ne'].lng)
+
+              const isWithinLongitude =
+                swLng <= neLng
+                  ? record.longitude >= swLng && record.longitude <= neLng
+                  : record.longitude >= swLng || record.longitude <= neLng
+
+              return isWithinLatitude && isWithinLongitude
+            }),
+          }
```

Comment:
This return statement could be easier to read. Can you break the filter function out into a separate function?